src/seq/nucleotide.jl

```diff
@@ -97,27 +97,27 @@ end
 
 Test if nucleotide is surely a purine.
 """
-function ispurine(nt::Nucleotide)
+@inline function ispurine(nt::Nucleotide)
     bits = reinterpret(UInt8, nt)
-    return bits != 0 && (bits & 0b1010) == 0
+    return (bits != 0) & ((bits & 0b1010) == 0)
```

Comment:
If no performance gain is detectable, I feel like keeping it as is to minimize the difference from the master. Changing it may send a wrong signal to readers that this change is based on evidence. We can replace them with non-short-circuit operators anytime once benchmark supports that.